### PR TITLE
feat(intelligence): event_id + confidence in search responses, badge in smart-search header (Sprint 1 UI, PR-A+B)

### PR DESCRIPTION
**Pull request by alex-budanov**
First UI slice for the AI-observability work shipped in bugspotter-intelligence#34. Two thin, independently-shippable pieces combined into one PR because they share the response-shape change.

**PR-A** — thread `event_id` (and an overall `confidence` on `SearchResponse`) through the intelligence response TS types on both backend (`packages/backend/src/services/intelligence/types.ts`) and admin (`apps/admin/src/types/intelligence.ts`). Only added on response types where the upstream intelligence service actually emits the field post-Sprint-1: `Search`, `Ask`, `EnrichBug`. `SimilarBug` / `MitigationResponse` are intentionally left alone — those call sites aren't wrapped in `record_generate` upstream yet, so typing them would advertise a field that's always `null`.

**PR-B** — new `ConfidenceBadge` ([apps/admin/src/components/intelligence/confidence-badge.tsx](apps/admin/src/components/intelligence/confidence-badge.tsx)): three states (`<0.6` yellow "Needs review", `[0.6, 0.85)` muted "AI-suggested", `≥0.85` hidden). Rendered once in the smart-search header next to the existing "cached" chip ([semantic-search-bar.tsx](apps/admin/src/components/bug-reports/semantic-search-bar.tsx)). Hidden on cache hits since the confidence belongs to the original LLM call, not the cached read. Thresholds hard-coded per the [intelligence design doc](https://github.com/apex-bridge/bugspotter-intelligence/blob/main/docs/ai-observability-design.md); per-org tuning is a follow-up.

Tests: 10 new unit tests for `ConfidenceBadge` (null/undefined/NaN, three thresholds + bounds, tooltip percentage). i18n keys added to en/ru/kk (`pnpm validate:i18n` clean). Admin: 953 passed, build clean. Backend: 2928 passed, typecheck clean.

Out of scope: the 👍/👎 feedback wiring (PR-C) and the admin observability dashboard (PR-D) — both depend on the `event_id` field this PR threads through, so they unblock as soon as this lands.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added AI confidence indicators to search results in smart mode: "Needs review", "AI-suggested", or no badge for high-confidence results.

* **Documentation**
  * Added localized strings for confidence badges in English, Kazakh, and Russian.

* **Tests**
  * Added comprehensive tests covering badge rendering, thresholds, tooltip percentages, and boundary cases.

* **Chores**
  * Intelligence responses now include optional confidence and event identifier metadata for smart-mode results.

### CODE REVIEW

File: apps/admin/src/components/intelligence/confidence-badge.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      variant="outline"
+      className={`gap-1 text-gray-600 ${className ?? ''}`}
+      title={t('intelligence.confidence.aiSuggestedTooltip', {
+        score: Math.round(confidence * 100),
+      })}
+    >
+      <Sparkles className="h-3 w-3" aria-hidden="true" />
+      {t('intelligence.confidence.aiSuggested')}
+    </Badge>
+  );
```
[the remark this answers is not in the document]
Comment (alex-budanov) [reply apps/admin/src/components/intelligence/confidence-badge.tsx]: Good catch — fixed in 9d88841. Gating now uses the same rounded integer that goes into the tooltip (`Math.round(confidence * 100)` vs `NEEDS_REVIEW_PCT = 60` / `HIGH_PCT = 85`), so the badge category and the displayed percentage can't contradict each other. 4 new tests pin the boundary cases (0.594 / 0.596 / 0.844 / 0.846).

**Comment by alex-budanov** [comment]
Declining the defensive [0, 1] range check in ConfidenceBadge from the CodeRabbit review: upstream API guarantees the range, CodeRabbit itself flagged it "💤 Low value", and out-of-range values render harmlessly today (negative → yellow, > 1 → hidden). Adding the check would be guarding a scenario that can't happen, against the project's "don't validate scenarios that can't happen" guidance.

**Comment by alex-budanov** [comment]
@claude review

File: apps/admin/src/components/intelligence/confidence-badge.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  /** Optional extra tailwind classes (e.g. spacing in the parent layout). */
+  className?: string;
+}
+
+export function ConfidenceBadge({ confidence, className }: ConfidenceBadgeProps) {
+  const { t } = useTranslation();
+
+  if (typeof confidence !== 'number' || Number.isNaN(confidence)) {
+    return null;
+  }
```
[the remark this answers is not in the document]
Comment (alex-budanov) [reply apps/admin/src/components/intelligence/confidence-badge.tsx]: Accepted in 250391e. Upstream JSON can't actually encode Infinity (the case stays theoretical from the API side), but -Infinity is a real footgun if it ever surfaces: `Math.round(-Infinity * 100) = -Infinity` would fall into the "Needs review" branch and render a `-Infinity%` tooltip. `!Number.isFinite` tightens the contract at zero cost. Two new tests cover ±Infinity.

File: apps/admin/src/components/intelligence/confidence-badge.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  const score = Math.round(confidence * 100);
+
+  if (score >= HIGH_PCT) {
+    return null;
+  }
+  if (score < NEEDS_REVIEW_PCT) {
+    return (
+      <Badge
+        variant="outline"
+        className={`gap-1 border-amber-300 bg-amber-50 text-amber-900 ${className ?? ''}`}
```
[the remark this answers is not in the document]
Comment (alex-budanov) [reply apps/admin/src/components/intelligence/confidence-badge.tsx:41]: Declining: the surrounding context is light-mode-only — the underlying `Badge` UI primitive (apps/admin/src/components/ui/badge.tsx), `semantic-search-bar.tsx`'s purple panel header (`bg-purple-50 border-purple-100`), and the existing `deflection-stats-card.tsx` all have no `dark:` variants. Adding dark-mode here only would be inconsistent. The right shape is a separate "dark-mode across the intelligence surface" change.

File: apps/admin/src/components/intelligence/confidence-badge.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      >
+        <AlertTriangle className="h-3 w-3" aria-hidden="true" />
+        {t('intelligence.confidence.needsReview')}
+      </Badge>
+    );
+  }
+  return (
+    <Badge
+      variant="outline"
+      className={`gap-1 text-gray-600 ${className ?? ''}`}
```
[the remark this answers is not in the document]
Comment (alex-budanov) [reply apps/admin/src/components/intelligence/confidence-badge.tsx:52]: Declining for the same reason as the amber variant — the surrounding components are light-mode-only and a one-off `dark:` variant on this badge would clash with the panel it sits in. Separate change.

**Comment by alex-budanov** [comment]
Closing to restart the review thread. New PR coming off the same branch (feat/ai-observability-ui).
